[IOTDB-700]Feature/iotdb 700 add openid

JulianFeinauer opened the pull request:
As discussed with @jixuan1989 in Slack he will have a look on this Branch and we will discuss how to integrate multiple Auth Providers in the config.

xingtanzjr commented on server/src/main/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizer.java:
```diff
@@ -0,0 +1,248 @@
+            throw new IllegalArgumentException("OpenID Connect Provider URI must be given!");
+        }
+
+        //
```
Let's remove empty comment here : )

jixuan1989 replied:
welcome back @xingtanzjr

jixuan1989 commented on server/src/main/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizer.java:
```diff
@@ -19,44 +19,32 @@
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import com.nimbusds.oauth2.sdk.ParseException;
```
the unused "import" should be removed..

jixuan1989 commented on server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java:
```diff
@@ -0,0 +1,64 @@
+        String jwt = "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJxMS1XbTBvelE1TzBtUUg4LVNKYXAyWmNONE1tdWNXd25RV0tZeFpKNG93In0.eyJleHAiOjE1OTAzMTcxNzYsImlhdCI6MTU5MDMxNjg3NiwianRpIjoiY2MyNWQ3MDAtYjc5NC00OTA4LTg0OGUtOTRhNzYzNmM5YzQxIiwiaXNzIjoiaHR0cDovL2F1dGguZGVtby5wcmFnbWF0aWNpbmR1c3RyaWVzLmRlL2F1dGgvcmVhbG1zL0lvVERCIiwiYXVkIjoiYWNjb3VudCIsInN1YiI6Ijg2YWRmNGIzLWE4ZTUtNDc1NC1iNWEwLTQ4OGI0O ...
+
+        OpenIdAuthorizer authorizer = new OpenIdAuthorizer(JSONObjectUtils.parse(OPEN_ID_PUBLIC_JWK));
+        boolean login = authorizer.login(jwt, null);
```
has to consider how to write the Test. 
Either use a Mock or use a stable OpenID server.

JulianFeinauer commented:
@jixuan1989 I added some english users documentation and fixed the sonar issues (the last I think is fine, better log more than less). So I am open for any further feedback!

jixuan1989 commented:
Hi @JulianFeinauer , pls pay attention to then the Travis error  (using WinOS, some resource is not released when the JWT connection fails. If we do not fix the issue, then all Travis after merging this PR will fail...)

jixuan1989 commented:
As for the sonar issue about the log, I think following Sonar tips is fine.

As you have already thrown the exception to the caller. The handler who receives the exception should handle the exception or print the exception. 

But in this case, the exception log may be printed twice.

![image](https://user-images.githubusercontent.com/1021782/82964386-ecfd8e80-9ff7-11ea-9c10-74c49a690dac.png)

JulianFeinauer commented:
@jixuan1989 I fixed everything. I hope you are happy with me now? :)

jixuan1989 commented:
Hi @JulianFeinauer , Look good to me. 
I will add the Chinese user guide later.
How about add this feature into v0.10.
